descwl_shear_sims/galaxies.py

```diff
@@ -696,6 +696,14 @@ def __init__(
             select_lower_limit=select_lower_limit,
             select_upper_limit=select_upper_limit,
         )
+        ngal = len(self._wldeblend_cat)
+        if "prob" in self._wldeblend_cat.dtype.names and ngal > 0:
```

Comment:
This field is not in the standard file OneSqDeg.fits

This creates a situation where the code would silently do something different based on the file data, but it is still internally assuming OneSqDeg.fits is still the name, which is confusing and could cause bugs.

I think maybe this should not be part of this class. I think it makes more sense for you to create your own galaxy class that expects a different input file.